### DISCUSSION TITLE
Teleport agent integration with Vault

### DISCUSSION THREAD

**Discussion by igorrudyk** [gravitational/teleport#26428] [Q&A]
Hello guys,
Kindly ask you for assistance.

Problem: After Teleport server redeployment, Teleport agents could not connect with a predefined token. As result, Teleport agents are disconnected.
There is a workaround to delete the file "/var/lib/teleport/proc/*" & restart teleport service - to regenerate certificates.

Is there an option to integrate the Teleport agent with Vault to avoid a manual workaround?

**Comment by stevenGravy** [comment]
If you are in the AWS or Azure env would suggest taking a look at the options here https://goteleport.com/docs/management/join-services-to-your-cluster/.

**Reply by igorrudyk** [reply]
Hello @stevenGravy ,
Teleport Server is being deployed on the GKE cluster via Helm chart. And Teleport Agents are connected via a Predefined token. 

But the problem is that after Teleport Server recreation/outage, a Predefined token is added to Teleport Server, but still Teleport Agents cannot reconnect to the server. 
And only cleaning the contents of /var/lib/teleport/proc inside the Agents and then restarting the service should fix it and force a join with a new certificate.

**Reply by stevenGravy** [reply]
What's going on to cause recreation? Teleport is designed to be upgraded and recover from issues.

**Reply by igorrudyk** [reply]
@stevenGravy , in other words, if we remount PV from the previous Teleport set-up - everything will be working fine? 
And do not need to remove /var/lib/teleport/proc inside the Agents and restart the Teleport Agent service? Correct?

**Reply by stevenGravy** [reply]
Yes, it should rejoin.
